cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2874,33 +2874,6 @@ endmacro()
 # ----------------------------------------------------------------------
 # Dependencies for Arrow Flight RPC
 
-macro(ensure_absl)
-  if(NOT absl_FOUND)
-    if(${absl_SOURCE} STREQUAL "AUTO")
-      # We can't use resolve_dependency(absl 20211102) to use Abseil
-      # 20211102 or later because Abseil's CMake package uses "EXACT"
```

Comment:
`resolve_dependency` seems to be quite sophisticated these days and it only passes a version if the `resolve_dependency` call is inflexible, so this workaround is not needed anymore (?).

@kou what you think?

Reply:
Yes. `resolve_dependency(FORCE_ANY_NEWER_VERSION TRUE)` should work.